### PR TITLE
[Fix] `jsx-curly-brace-presence`: do not trigger on strings containing a quote character

### LINKED ISSUE #3795: [Bug]: `jsx-curly-brace-presence` doesn't trigger on strings containing a quote character
### Is there an existing issue for this?

- [X] I have searched the existing issues and my issue is unique
- [X] My issue appears in the command-line and not only in the text editor

### Description Overview

The curly braces are unnecessary here and should be reported by the rule. If the single quote is replaced with any other character, the rule does trigger.
```jsx
<Foo
  bar={"'"}
/>
```

### Expected Behavior

Report an error with a suggested fix to
```jsx
<Foo
  bar="'"
/>
```

### eslint-plugin-react version

4.3.1

### eslint version

8.57.0

### node version

20.14.12

### PR DESCRIPTION
**Pull request by akulsr0**
Fixes #3795

### CODE REVIEW

File: tests/lib/rules/jsx-curly-brace-presence.js [hunk 1]
```diff
@@ -934,6 +934,13 @@ ruleTester.run('jsx-curly-brace-presence', rule, {
       errors: [{ messageId: 'unnecessaryCurly' }],
       options: [{ props: 'never', children: 'never', propElementValues: 'never' }],
       features: ['no-ts'],
+    },
+    {
+      code: `<Foo bar={"'"} />`,
+      output: `<Foo bar="'" />`,
+      errors: [{ messageId: 'unnecessaryCurly' }],
+      options: [{ props: 'never', children: 'never', propElementValues: 'never' }],
+      features: ['no-ts'],
```
Comment (ljharb) [inline tests/lib/rules/jsx-curly-brace-presence.js:943]: why does this need to omit TS tests?

Comment (akulsr0) [reply tests/lib/rules/jsx-curly-brace-presence.js]: missed while copy pasting, reverting

**Comment by dennisoelkers** [comment]
@akulsr0, @ljharb: This PR is introducing a regression for us, it is changing:

```
                                help={'The maximum time range for searches. (i.e. "P30D" for 30 days, "PT24H" for 24 hours)'}
```
to
```
                                help="The maximum time range for searches. (i.e. "P30D" for 30 days, "PT24H" for 24 hours)"
```

which is obviously an incorrect string.

**Comment by ljharb** [comment]
@dennisoelkers you’re right; could you file a new issue for that?

**Comment by dennisoelkers** [comment]
@ljharb: There you go: #3814

**Pull request merged**